File: Serenity Practice/HosbitalSerenity.Web/Modules/Hosbital/Speciality/SpecialityColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace HosbitalSerenity.Hosbital.Columns;

[ColumnsScript("Hosbital.Speciality")]
[BasedOnRow(typeof(SpecialityRow), CheckNames = true)]
public class SpecialityColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int SpecialityId { get; set; }
    [EditLink]
    public string SpecialityName { get; set; }
}

File: Serenity Practice/HosbitalSerenity.Web/Modules/ServerTypes/Hosbital/SpecialityColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SpecialityRow } from "./SpecialityRow";

export interface SpecialityColumns {
    SpecialityId: Column<SpecialityRow>;
    SpecialityName: Column<SpecialityRow>;
}

export class SpecialityColumns extends ColumnsBase<SpecialityRow> {
    static readonly columnsKey = 'Hosbital.Speciality';
    static readonly Fields = fieldsProxy<SpecialityColumns>();
}